app/cni/cmd/install/validation.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"github.com/pkg/errors"
+)
+
+func isValidConfFile(file string) bool {
+	parsed, err := parseFileToHashMap(file)
+	if err != nil {
+		log.Error(err, "could not unmarshal config file")
+		return false
+	}
+
+	configType, ok := parsed["type"]
+	if ok {
+		log.V(1).Info("config valid", "file", file, "type", configType)
+		return true
+	}
+	log.V(1).Info("config not valid", "file", file)
```

Comment:
why it's not valid? Again: this could be a part of error that you could return

Reply:
it's not valid because it does not contain "type" field, added that to the error message 👍